StudentList/Fragments/StudentListFragment.cs

```diff
@@ -1,39 +1,45 @@
 ﻿using System.Collections.Generic;
 using System.Globalization;
+using System.Threading.Tasks;
 using Android.OS;
 using Android.Support.V7.App;
 using Android.Support.V7.Widget;
 using Android.Views;
 using Android.Widget;
 using StudentList.Adapters;
-using StudentList.Model;
+using StudentList.Common.Dialogs;
+using StudentList.Extensions;
+using StudentList.Models;
 using StudentList.Providers.Interfaces;
 
 namespace StudentList.Fragments
 {
     public class StudentListFragment : Android.Support.V4.App.Fragment
     {
         private readonly StudentFilter studentFilter;
-        private readonly StudentAdapter studentAdapter;
 
         private RecyclerView recyclerView;
+        private StudentAdapter studentAdapter;
         private IStudentRepository repository;
 
+        private LoadingDialog loadingDialog;
         private TextView filteringResultTextView;
-        private ProgressBar loadingProgressBar;
         private TextView studentsCountTextView;
 
         public StudentListFragment(StudentFilter studentFilter)
         {
             this.studentFilter = studentFilter;
-            this.studentAdapter = new StudentAdapter();
         }
 
         public override void OnCreate(Bundle savedInstanceState)
         {
             base.OnCreate(savedInstanceState);
 
-            this.repository = new StudentsRepository(this.Activity);
+            this.studentAdapter = new StudentAdapter();
+            this.loadingDialog = new LoadingDialog(this.Context);
+            this.repository = new StudentsRepository(
+                this.Activity,
```

Comment:
You cant have a reference to activit inside your repo class (repo = pure .net class without references to android framework).